src/util.ts

```diff
@@ -18,36 +17,39 @@ export function isNegativeReaction(content: octokit.ReactionContent): boolean {
   return ((content === ReactionContent.ThumbsDown) || (content === ReactionContent.Confused));
 }
 
-export function containsPositiveReaction(comment: octokit.DiscussionCommentEdge): boolean {
+export function containsPositiveReaction(comment: DiscussionCommentEdge): boolean {
   return comment.node?.reactions.nodes?.some(reaction => {
     return isPositiveReaction(reaction?.content!);
   })!;
 }
 
-export function containsNegativeReaction(comment: octokit.DiscussionCommentEdge): boolean {
+export function containsNegativeReaction(comment: DiscussionCommentEdge): boolean {
   return comment.node?.reactions.nodes?.some(reaction => {
     return isNegativeReaction(reaction?.content!);
   })!;
 }
 
-export function hasReaction(comment: octokit.DiscussionCommentEdge): boolean {
+export function hasReaction(comment: DiscussionCommentEdge): boolean {
   return comment?.node?.reactions.nodes?.length !== 0;
 }
 
-export function containsText(comment: octokit.DiscussionCommentEdge, text: string): boolean {
+export function containsKeyword(comment: DiscussionCommentEdge, text: string): boolean {
   return comment?.node?.bodyText?.indexOf(text)! >= 0;
 }
 
-export function exceedsDaysUntilStale(comment: octokit.DiscussionCommentEdge, staleTimeDays: number): boolean {
+export function exceedsDaysUntilStale(comment: DiscussionCommentEdge, staleTimeDays: number): boolean {
   return (daysSinceComment(comment) >= staleTimeDays);
 }
 
-// TODO: Implement this function
-export function hasReply(comment: octokit.DiscussionCommentEdge, discussion: octokit.DiscussionEdge): boolean {
-  return true;
+export function hasReplies(comment: DiscussionCommentEdge): boolean{
+  return comment.node?.replies.edges?.some(reply => {
+    return (reply?.node?.bodyText.length !== 0);
+  })!;
 }
 
-// TODO: Implement this function
-export function hasInstructionsReply(comment: octokit.DiscussionCommentEdge, discussion: octokit.DiscussionEdge): boolean {
-  return true;
+
+export function hasInstructionsText(comments: DiscussionCommentConnection, INSTRUCTIONS_TEXT: string): boolean {
```

Comment:
I have declared 2 functions with slightly changed flow. When we start iterating over comments, I check whether there is any comment with InstructionText(hasInstructionsText()). If yes, that flag is enabled and set to true before the very beginning of processing comments.

Again when we loop thru comments, checking if any comment has the PROPOSED_ANSWER_KEYWORD. if it does have that, we check its replies by this function hasReplies(comment). Since we are getting comments metadata info from earlier step, we would know here whether the current comment has any replies(which has the keyword as well). On finding this , `attention` label is added.

I also make sure that InstructionText is not added multilple times so the above flag comes into picture.

If there s any doubt, we can discuss it offline.

Reply:
let's move this discussion to the other thread I just replied to